src/lib.rs

```diff
@@ -709,14 +727,19 @@ macro_rules! enum_impl {
             ),*
         }
 
+        /// # Safety
+        /// - This is implemented automatically by the bit-struct crate.
+        #[allow(clippy::use_self)]
         unsafe impl $crate::BitCount for $name {
             const COUNT: usize = $crate::bits($crate::count_idents!(0, [$($field),*]));
         }
 
+        #[allow(clippy::use_self)]
         impl $name {
             const VARIANT_COUNT: usize = $crate::enum_impl!(COUNT $fst_field $(,$field)*);
         }
 
+        #[allow(clippy::undocumented_unsafe_blocks, clippy::use_self)]
```

Comment:
I think it'd be better to document these with a safety comment than put a clippy allowance (unless you tried that and it didn't work?)

Reply:
It didn't seem to work for some odd reason. If you want to allocate time to try it and it works for you, I wouldn't mind. It seemed `clippy::use_self` also had false positives—I couldn't find anywhere I could use `Self,` but I wouldn't be surprised if I am just dumb.

Reply:
I was mainly referring to the `clippy::undocumented_unsafe_blocks` allowance, which you should be able to resolve by putting a safety comment on this `unsafe impl` and the others like it.

Also, with the `clippy::use_self`, it seems like you missed the spot where the usage is happening because I'm still seeing that lint trigger with the allowances you added in.